theano/gpuarray/c_code/dnn_fwd.c

```diff
@@ -309,7 +333,52 @@ APPLY_SPECIFIC(conv_fwd)(PyGpuArrayObject *input, PyGpuArrayObject *kerns,
       cuda_exit(c->ctx);
       return 1;
     }
+    // save worksize for next time/cache
+    prev_algo.wsSize = worksize;
 
+    // Add to the cache
+    if (params->choose_algo)
```

Comment:
If the user specify time_once of guess_once, it mean the cache isn't used. As normally, the algo is specified globally, what I wrote made sence.

But you raised a good point, in case people specify the algo per layer, the updating the cache make sense. To help us not forget that case, can you add a comment for this? Something like:

//update the cache, even if this node use *_once algo in case the user specify the algo per layer and not globally.